cpp/src/jit/row_ir.hpp

```diff
@@ -0,0 +1,429 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <cudf/ast/detail/operators.hpp>
+#include <cudf/ast/expressions.hpp>
+#include <cudf/io/types.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/export.hpp>
+
+#include <string>
+#include <string_view>
+
+namespace CUDF_EXPORT cudf {
+namespace detail {
+
+/// @brief cudf IR for operations acting on elements of a row
+namespace row_ir {
+
+/// @brief The target for which the IR is generated.
+enum class target {
+  CUDA = 0  /// < CUDA C++
+};
+
+/// @brief The type information of the variable used in the IR.
+struct type_info {
+  data_type type = data_type{type_id::EMPTY};  ///< The data type of the variable
+};
+
+/// @brief The information about the variable used in the IR.
+struct var_info {
+  std::string id = {};  ///< The variable identifier
+  type_info type = {};  ///< The type information of the variable
+};
+
+/// @brief The information about the variable used in the IR without type information. The type
+/// information is auto-deduced from an IR node.
+struct untyped_var_info {
+  std::string id = {};  ///< The variable identifier
+};
+
+/// @brief The information needed to instantiate the IR nodes
+struct instance_info {
+  std::span<var_info const> inputs;           ///< The input variables
+  std::span<untyped_var_info const> outputs;  ///< The output variables
+};
+
+/// @brief The information about the target for which the IR is generated.
+struct target_info {
+  target id = target::CUDA;  ///< The target identifier
+};
+
+/// @brief The context within which the IR is instantiated.
+/// This context is used to generate temporary variable identifiers and any state setup needed for
+/// the IR instantiation.
```

Comment:
Could you adopt the doxygen style from the documentation guidelines example?
https://github.com/rapidsai/cudf/blob/branch-25.10/cpp/doxygen/developer_guide/DOCUMENTATION.md#the-example